iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/helpers/X509CertificateGenerator.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    /**
+     * generate a new certificate using the provided common name
+     */
+    private void generateCertificate(String commonName) throws CertificateException, NoSuchAlgorithmException, OperatorCreationException, SignatureException, NoSuchProviderException, InvalidKeyException
+    {
+        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+        keyGen.initialize(1024, new SecureRandom());
+        KeyPair keypair = keyGen.generateKeyPair();
+        this.x509Certificate = createX509CertificateFromKeyPair(keypair, commonName);
+        this.x509Thumbprint = new String(Hex.encodeHex(DigestUtils.sha(x509Certificate.getEncoded())));
+        this.publicCertificate = getPublicCertificatePem(x509Certificate);
+        this.privateKey = getPrivateKeyPem(keypair.getPrivate());
+    }
+
+    /**
+     * @return the thumbprint for the the generated x509 certificate
+     */
+    public String getX509Thumbprint()
+    {
+        return x509Thumbprint;
+    }
+
+    /**
+     * @return the complete x509 certificate that was generated
+     */
+    public X509Certificate getX509Certificate()
+    {
+        return x509Certificate;
+    }
+
+    /**
+     * @return the Pem formatted public certificate for the generated certificate key pair
+     */
+    public String getPublicCertificate()
+    {
+        return publicCertificate;
+    }
+
+    /**
+     * @return the Pem formatted private key for the generated certificate key pair
+     */
+    public String getPrivateKey()
+    {
+        return privateKey;
+    }
+
+    /**
+     * Create a new self signed x509 certificate without specifying a common name
+     */
+    private static X509Certificate createX509CertificateFromKeyPair(KeyPair keyPair) throws CertificateException, NoSuchAlgorithmException, OperatorCreationException, SignatureException, NoSuchProviderException, InvalidKeyException
+    {
+        return createX509CertificateFromKeyPair(keyPair, null);
+    }
+
+    /**
+     * Create a new self signed x509 certificate with the specified common name
+     */
+    private static X509Certificate createX509CertificateFromKeyPair(KeyPair keyPair, String commonName)
+            throws OperatorCreationException, CertificateException, InvalidKeyException, NoSuchAlgorithmException,
+            NoSuchProviderException, SignatureException
+    {
+        String issuerString = "C=US, O=Microsoft, L=Redmond, OU=Azure";
+        // subjects name - the same as we are self signed.
+        String subjectString = "C=US, O=Microsoft, L=Redmond, OU=Azure";
+
+        if (commonName != null && !commonName.isEmpty())
+        {
+            issuerString += ", CN=" +commonName;
+            subjectString += ", CN=" +commonName;
+        }
+
+        X500Name issuer = new X500Name( issuerString );
+        BigInteger serial = BigInteger.ONE;
+        Date notBefore = new Date();
+        Date notAfter = new Date( System.currentTimeMillis() + ( 60*60*1000 ) );
```

Comment:
Please use `TimeUnits`

Reply:
I don't follow what you want here. Date constructors do not take Timeunits, and the JcaX509v3CertificateBuilder constructor requires Date objects for defining when the certificate lives for

Reply:
`Date notAfter = new Date( System.currentTimeMillis()+TimeUnit.HOURS.toMillis(1))`

Reply:
Got it, I can do that